assignments/lambda-classes.js

```diff
@@ -1 +1,98 @@
 // CODE here for your Lambda Classes
+class Person {
+    constructor(parentAttributes) {
+        this.name = parentAttributes.name;
+        this.age = parentAttributes.age;
+        this.location = parentAttributes.location;
+        this.gender = parentAttributes.gender;
+    }
+    speak() {
+        return `Hello my name is ${this.name}, I am from ${this.location}`;
+    }
+}
+
+class Instructor extends Person {
+    constructor(instructorAttributes) {
+        super(instructorAttributes);
+        this.specialty = instructorAttributes.specialty;
+        this.favLanguage = instructorAttributes.favLanguage;
+        this.catchPhrase = instructorAttributes.catchPhrase;
+    }
+    demo(subject) {
+        return `Today we are learning about ${subject}`;
+    }
+    grade(student, subject) {
+        return `${student.name} receives a perfect score on ${subject}`;
+    }
+}
+
+class Student extends Person {
+    constructor(studentAttributes) {
+        super(studentAttributes);
+        this.previousBackground = studentAttributes.previousBackground;
+        this.className = studentAttributes.className;
+        this.favSubjects = studentAttributes.favSubjects;
+    }
+    listsSubjects() {
+        return this.favSubjects;
```

Comment:
`favSubjects` is an array, so you're going to want to iterate over the array's elements.